src/solvers/refinement/string_constraint_generator_indexof.cpp

```diff
@@ -116,70 +116,35 @@ exprt string_constraint_generatort::add_axioms_for_index_of_string(
     equal_exprt(haystack[plus_exprt(qvar, offset)], needle[qvar]));
   axioms.push_back(a3);
 
-  if(!is_constant_string(needle))
-  {
-    // string_not contains_constraintt are formulas of the form:
-    // forall x in [lb,ub[. p(x) => exists y in [lb,ub[. s1[x+y] != s2[y]
-    string_not_contains_constraintt a4(
-      from_index,
-      offset,
-      contains,
-      from_integer(0, index_type),
-      needle.length(),
-      haystack,
-      needle);
-    axioms.push_back(a4);
-
-    string_not_contains_constraintt a5(
-      from_index,
+  // string_not contains_constraintt are formulas of the form:
+  // forall x in [lb,ub[. p(x) => exists y in [lb,ub[. s1[x+y] != s2[y]
+  string_not_contains_constraintt a4(
+    from_index,
+    offset,
+    contains,
+    from_integer(0, index_type),
+    needle.length(),
+    haystack,
+    needle);
+  axioms.push_back(a4);
+
+  string_not_contains_constraintt a5(
+    from_index,
+    plus_exprt(
       minus_exprt(haystack.length(), needle.length()),
-      not_exprt(contains),
-      from_integer(0, index_type),
-      needle.length(),
-      haystack,
-      needle);
-    axioms.push_back(a5);
-  }
-  else
-  {
-    // Unfold the existential quantifier as a disjunction in case of a constant
-    // a4 && a5 <=> a6:
-    //  forall n:[from_index,|haystack|-|needle|].
-    //    !contains || n < offset ==>
-    //      haystack[n] != needle[0] || ... ||
-    //      haystack[n+|needle|-1] != needle[|needle|-1]
-    symbol_exprt qvar2=fresh_univ_index("QA_index_of_string_2", index_type);
-    mp_integer sub_length;
-    INVARIANT(
-      !to_integer(needle.length(), sub_length),
-      string_refinement_invariantt("a constant string must have constant "
-        "length"));
-    exprt::operandst disjuncts;
-    for(mp_integer offset=0; offset<sub_length; ++offset)
-    {
-      exprt expr_offset=from_integer(offset, index_type);
-      plus_exprt shifted(expr_offset, qvar2);
-      disjuncts.push_back(
-        not_exprt(equal_exprt(haystack[shifted], needle[expr_offset])));
-    }
-
-    or_exprt premise(
-      not_exprt(contains), binary_relation_exprt(qvar2, ID_lt, offset));
-    minus_exprt length_diff(haystack.length(), needle.length());
-    string_constraintt a6(
-      qvar2,
-      from_index,
-      plus_exprt(from_integer(1, index_type), length_diff),
-      premise,
-      disjunction(disjuncts));
-    axioms.push_back(a6);
-  }
+      from_integer(1, index_type)),
```

Comment:
Presumably this is to create an exclusive upper-bound? If so put a quick end-of-line comment saying so